src/Compression/CompressedWriteBuffer.cpp

```diff
@@ -50,9 +50,25 @@ CompressedWriteBuffer::CompressedWriteBuffer(
 
 CompressedWriteBuffer::~CompressedWriteBuffer()
 {
-    /// FIXME move final flush into the caller
-    MemoryTracker::LockExceptionInThread lock;
-    next();
+    if (hasPendingData())
+    {
+
+#ifndef NDEBUG
+        /// No exception, buffer was not flushed. Consider it as a logical error.
+        if (std::uncaught_exception() == 0)
```

Comment:
It is better to use `std::uncaught_exceptions` here, since `std::uncaught_exception` is deprecated.
And also this check is not ideal, since `CompressedWriteBuffer` can be created during unwinding.

By the way, why only one buffer had been changed? (And not all buffers in general)

Reply:
> since CompressedWriteBuffer can be created during unwinding

Hopefully this never happens. If not, I am ok to see it in debug.